### ISSUE TITLE
Bug: Editor fails to load bases and deletes them on save

### ISSUE THREAD

**Issue by antonelli** [goatfungus/NMSSaveEditor#195]
The editor currently fails to display my bases (it still displays base storage containers, but not the bases themselves). The dropdown is empty for my 3 bases.

When saving after making any changes, the bases in-game are then all gone.

The only oddity I can think of with my bases is I use numbers and periods in their names (`Base.3.Farm.CircuitBoards` for example), so that may be causing a parse issue. If I have time later, I'll try renaming all my bases, but I also name my systems/planets this way so that may not be a fix if you parse through those as well for the bases.

Here's the log:

```
[DEBUG] Java Vendor: Oracle Corporation
[DEBUG] Java Version: 1.8.0_221
[DEBUG] Working Dir: C:\Users\[me]\AppData\Roaming\HelloGames_SaveEditor
[INFO] Starting Editor...
[DEBUG] Mem Usage: 16/260 MB
[DEBUG] Latest version: "1.6.2"
[DEBUG] Current version: "1.6.2"
[INFO] Loading slots: C:\Users\[me]\AppData\Roaming\HelloGames\NMS\[me]
[INFO] Moving backup files...
[DEBUG] Mapping not available: tiH
[DEBUG] Save Path: C:\Users\[me]\AppData\Roaming\HelloGames\NMS\[me]
[INFO] Loading saves for Slot 3...
[INFO] Reading save file...
[INFO] Slot: Slot 3 - NORMAL
[INFO] Filename: save5.hg
[INFO] Parsing JSON...
[INFO] using name mappings: NMS 2.0
[WARNING] name mapping not found: Bur
[INFO] Finished.
```

**Comment by goatfungus** [comment]
If the dropdown is empty, then the UI part of the editor isn't able to read the bases at all, so won't be able to alter them. What I find interesting though is that your log doesn't show any edits to the file let alone any actual save. You should see a "[INFO] Writing save file..." line as well as a bunch of others.

If you've lost your bases and the save editor didn't actually change anything, then this would be a problem with the game itself.

**Comment by antonelli** [comment]
As I was taking screenshots for the below response, on a whim I went to my base computer and it has me "logging in as Guest", so it seems I've lost ownership of my bases. Which confirms I'm a victim of the bug I talk about in the response.

Sorry for this bug report, turns out it's a fault on their side and this annoying nexus teleport bug.

------

Sorry, I didn't run the save for that log I copied, here's a new one, but it has no additional warnings/errors.

I can confirm that it's the editor that's removing the bases. Here's how I reproduced:
- In-game, save and close in my base.
- In the editor, load the save and change something arbitrary (in this case, I maxed an item's stack size in a base storage container).
- Save in the editor.
- Load the in-game save, the base is gone.
- Reload the backup save, the base is still there.

There's still a very good chance that it's not the editor's fault. I was just reading about folks experiencing a similar bug to what I'm experiencing in-game. My capital freighter vanished after using the nexus teleporter (which is what prompted me to use the editor, to look at maybe getting it back). Other folks that reported their freighter disappearing in this instance also reported their bases getting bugged or missing. I wouldn't be surprised if my game's base data is screwed up in some way from that bug.

I used the editor's json view to confirm that all my items in my freighter still exist, but the freighter data itself is gone. Also in the json view I noticed the PersistentPlayerBases array is empty (assuming this is the base list).

Sounds like I'm an edge case caused by the bug, so this issue can probably be ignored and closed.

```
[DEBUG] Java Vendor: Oracle Corporation
[DEBUG] Java Version: 1.8.0_221
[DEBUG] Working Dir: C:\Users\[me]\AppData\Roaming\HelloGames_SaveEditor
[INFO] Starting Editor...
[DEBUG] Save Path: C:\Users\[me]\AppData\Roaming\HelloGames\NMS\[me]
[DEBUG] Mapping not available: tiH
[DEBUG] Mem Usage: 31/260 MB
[DEBUG] Latest version: "1.6.2"
[DEBUG] Current version: "1.6.2"
[INFO] Loading saves for Slot 3...
[INFO] Reading save file...
[INFO] Slot: Slot 3 - NORMAL
[INFO] Filename: save5.hg
[INFO] Parsing JSON...
[INFO] using name mappings: NMS 2.0
[WARNING] name mapping not found: Bur
[INFO] Finished.
[INFO] Formatting JSON...
[INFO] using name mappings: NMS 2.0
[INFO] Creating backup...
[INFO] Writing new save file...
[INFO] Finished.
[INFO] Loading saves for Slot 3...
[INFO] Finished.
```

Screen shot of the empty base list:
![Capture](https://user-images.githubusercontent.com/91856/63599465-aaea7c80-c58f-11e9-866f-183566f3e020.PNG)

After save:
![20190823102438_1](https://user-images.githubusercontent.com/91856/63600138-00735900-c591-11e9-888f-53741796c539.jpg)

Backup save:
![20190823102632_1](https://user-images.githubusercontent.com/91856/63600156-06693a00-c591-11e9-9310-fc2affd58932.jpg)
